avaframe/com1DFAPy/com1DFA.py

```diff
@@ -1707,6 +1715,58 @@ def readPartFromPickle(inDir, flagAvaDir=False):
     return Particles, TimeStepInfo
 
 
+def savePartToCsv(particleProperties, dictList, outDir):
+    """ Save each particle dictionary from a list to a csv file; works also for one dictionary instead of list
+
+        Parameters
+        ---------
+        dictList: list or dict
+            list of dictionaries or single dictionary
+    """
+
+    # set output directory
+    outDir = os.path.join(outDir, 'particlesCSV')
+    fU.makeADir(outDir)
+
+    # read particle properties to be saved
+    particleProperties = particleProperties.split('|')
+
+    # write particles locations and properties to csv file
+    nParticles = len(dictList)
+    count = 0
+    for m in range(nParticles):
+        particles = dictList[count]
+        x = particles['x'] + particles['xllcenter']
+        y = particles['y'] + particles['yllcenter']
+        z = particles['z']
+        simName = particles['simName']
+        partPropDict = {}
+
+        for partProp in particleProperties:
+            if partProp == 'velocityMagnitude':
+                ux = particles['ux']
+                uy = particles['uy']
+                uz = particles['uz']
+                partPropDict[partProp] = DFAtls.norm(ux, uy, uz)
+            else:
+                partPropDict[partProp] = particles[partProp]
+
+
+        with open(os.path.join(outDir, 'particles%s.csv.%d' % (simName, count)), 'w') as pFile:
```

Comment:
why the number after the extension? .cvs.2 for example

Reply:
this is for the time steps

Reply:
this is for adding all the time steps

Reply:
so the 2 corresponds to t = 2 sec?

Reply:
no to the second time step however in the file there is the parameter t for the time step